### ISSUE TITLE
Tune the initial distribution model parameters

### ISSUE THREAD

**Issue by MazenAmria** [Mouafak-Khaled/Flow-Q-Learning#57]
Try increasing the hidden layers (currently we have only 1 hidden layer, try using 2, no more than that), or increasing/decreasing the hidden dimension. During this tuning, you need to fix the latent dimension as 2, and then increase the steps until the model have time to converge (loss curve flattens). You may need to change the weights of the joint loss:

```py
loss = (w * reconstruction_loss + kl) / (w + 1)
```

to make sure that the trainer focuses more on the reconstruction of the observations. The KL is just a regularizer, so it's expected to increase a little bit but not too much (for 2D latent space if it saturated around ~1 then that's good enough I think). Additionally, you need to encode the observations and plot the heatmap of the density in latent space (similar to the KDE plot that was generated with PCA, but this time we don't need PCA as the encoder projects it to 2D), and qualitatively assess the distribution to make sure that it doesn't collapse to a single gaussian.

After that, scale the latent dimension to see if performs better.
